trinity/main.py

```diff
@@ -206,27 +242,62 @@ def trinity_boot(args: Namespace,
     networking_process.start()
     logger.info("Started networking process (pid=%d)", networking_process.pid)
 
+    main_endpoint.subscribe(
```

Comment:
Here is where we listen for the `ShutdownRequest` and then call `kill_trinity_gracefully`. Sidenote about lahja: You could also say `main_endpoint.stream(ShutdownRequest)` to receive the event but this is an async API that needs to be awaited and hence will introduce async/await creep. The `subscribe` API is perfect for these cases as it gives you a callback based way to still receive events asynchronously without introducing async/await creep. I'm not saying I encourage that usage over the other (as I quite like `async` / `await`) but it is handy to use in situations such as here.